src/fragmenter.c

```diff
@@ -114,28 +107,23 @@ static int tokenizerCallback(void *ctxPtr, const Token *tokInfo) {
   // Don't care about this token
   if (!term) {
     fragList->numToksSinceLastMatch++;
-    return 0;
+    return;
   }
 
-  Fragment *curFrag = FragmentList_AddMatchingTerm(fragList, termId, tokInfo->pos, tokInfo->raw,
-                                                   tokInfo->rawLen, term->score);
-
-  return 0;
+  FragmentList_AddMatchingTerm(fragList, termId, tokInfo->pos, tokInfo->raw, tokInfo->rawLen,
+                               term->score);
 }
 
 void FragmentList_FragmentizeBuffer(FragmentList *fragList, const char *doc, Stemmer *stemmer,
                                     StopWordList *stopwords, const FragmentSearchTerm *terms,
                                     size_t numTerms) {
   fragList->doc = doc;
   fragList->docLen = strlen(doc);
-  bufTokenizerCtx ctx = {.fragList = fragList, .terms = terms, .numTerms = numTerms};
-
-  tokenize(doc, &ctx, tokenizerCallback, stemmer, 0, stopwords, TOKENIZE_NOMODIFY);
-  const Fragment *frags = FragmentList_GetFragments(fragList);
-
-  float totalScore = 0;
-  if (fragList->numFrags) {
-    return;
+  RSTokenizer *tokenizer = NewSimpleTokenizer(stemmer, stopwords, TOKENIZE_NOMODIFY);
```

Comment:
can't we pool this tokenizer?

Reply:
We can, but this function is only used in tests